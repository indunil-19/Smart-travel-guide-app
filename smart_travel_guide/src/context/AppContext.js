import React, {createContext, useReducer} from 'react';
import { initialState, reducer as AppReducer } from '../reducer/AppReducer';

export const AppContext =createContext()

export const AppProvider=({children})=>{
    const [state, dispatch]= useReducer(AppReducer, initialState)

    return(
        <AppContext.Provider  value={{state,dispatch}}>
                {children}
        </AppContext.Provider>
    )
}