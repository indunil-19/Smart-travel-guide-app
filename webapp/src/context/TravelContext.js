import React, {createContext, useReducer} from 'react';
import { initialState, Travelreducer } from '../reducer/TravelReducer';

const TravelContext =createContext()

const TravelProvider=({children})=>{
    const [state, dispatch]= useReducer(Travelreducer, initialState)

    return(
        <TravelContext.Provider  value={{state,dispatch}}>
                {children}
        </TravelContext.Provider>
    )
}

export  {TravelContext,  TravelProvider}
 