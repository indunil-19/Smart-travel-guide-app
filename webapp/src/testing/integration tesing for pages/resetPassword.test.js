import React from "react"
import {render,screen, cleanup,waitFor, fireEvent} from "@testing-library/react";
import renderer from "react-test-renderer";
import ReactDOM from "react-dom";
import { MemoryRouter } from "react-router-dom";
import {TravelProvider} from "../../context/TravelContext"
import "@testing-library/jest-dom/extend-expect";
import { rest } from "msw";
import { setupServer } from "msw/node";
import 'whatwg-fetch'
import ResestPassword from "../../screens/TravelPlanApp/resestPassword";


describe('reset password plans page test', () => {
    const server=setupServer(
        rest.get("/reset-password",(req,res,ctx)=>{
            return res(
                ctx.json({
                   message:"check your email"
                })
            )
        })
    )
    beforeEach(()=>{      
    })
    beforeAll(() => server.listen());
    afterEach(() => server.resetHandlers());
    afterAll(() => server.close());
    afterEach(cleanup)

    it("shoud render without crashing", ()=>{
        const div = document.createElement("div");
        ReactDOM.render(<MemoryRouter><TravelProvider><ResestPassword /></TravelProvider></MemoryRouter>,div)
        
    })

    it("shoud render reset page correctly",async()=>{
        const div = document.createElement("div");
        render(<MemoryRouter><TravelProvider><ResestPassword /></TravelProvider></MemoryRouter>,div)
        fireEvent.change(screen.getByPlaceholderText("email"), {target:{value:"udayangana1998@gmail.com"}})
        fireEvent.click(screen.getByTestId("submit"))
        await waitFor(()=>{
            const toasts=screen.getAllByRole("list")
            expect(toasts[2]).toBeInTheDocument();
            
        })
    })

    it("should match snapshot",  () => {
        const tree = renderer.create(<MemoryRouter><TravelProvider><ResestPassword/></TravelProvider></MemoryRouter>);
        expect(tree).toMatchSnapshot();
    });


     
})
