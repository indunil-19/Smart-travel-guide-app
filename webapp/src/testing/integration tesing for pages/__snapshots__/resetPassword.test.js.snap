// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reset password plans page test should match snapshot 1`] = `
<div
  className="css-9w5vj"
>
  <p
    className="chakra-text css-1tapsqs"
  >
    Reset Password
  </p>
  <input
    className="chakra-input css-0"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="email"
    type="text"
    value=""
  />
  <button
    className="chakra-button css-1pgfz3"
    data-testid="submit"
    onClick={[Function]}
    type="button"
  >
    reset password
  </button>
</div>
`;
